### PR TITLE
add collector for networkpolicies

### PR DESCRIPTION
**Pull request by tariq1890**

### CODE REVIEW

**Comment by zouyee** [comment]
As with other PRs, what is your scenario with these metrics?  May need to give a rough description.

File: internal/collector/builder.go [hunk 1]
```diff
@@ -124,9 +125,10 @@ var availableCollectors = map[string]func(f *Builder) *coll.Collector{
 	"jobs":                     func(b *Builder) *coll.Collector { return b.buildJobCollector() },
 	"limitranges":              func(b *Builder) *coll.Collector { return b.buildLimitRangeCollector() },
 	"namespaces":               func(b *Builder) *coll.Collector { return b.buildNamespaceCollector() },
+	"networkpolicies":          func(b *Builder) *coll.Collector { return b.buildNetworkPolicyCollector() },
 	"nodes":                    func(b *Builder) *coll.Collector { return b.buildNodeCollector() },
 	"persistentvolumeclaims":   func(b *Builder) *coll.Collector { return b.buildPersistentVolumeClaimCollector() },
-	"persistentvolumes":        func(b *Builder) *coll.Collector { return b.buildPersistentVolumeCollector() },
+	"persistexntvolumes":       func(b *Builder) *coll.Collector { return b.buildPersistentVolumeCollector() },
```
Comment (brancz) [inline internal/collector/builder.go]: this seems like it was a mistake

**Comment by tariq1890** [comment]
No specific use case I have in mind. I just thought that there would be a potential use for tracking `networkPolicy` between the Pod communications as kube-state-metrics is commonly used Pod metrics collection and analysis.

**Comment by brancz** [comment]
I think I'm happy to add it, if we have a concrete use case.

**Pull request closed without merging**